src/transforms/filter.js

```diff
@@ -54,6 +63,16 @@ exports.supplyDefaults = function(transformIn, fullData, layout) {
     coerce('value');
     coerce('filtersrc');
 
+    // numeric values as character should be converted to numeric
+    if(Array.isArray(transformOut.value)) {
+        transformOut.value = transformOut.value.map(function(v) {
+            if(isNumeric(v)) v = +v;
+            return v;
+        });
+    } else {
+        if(isNumeric(transformOut.value)) transformOut.value = +transformOut.value;
```

Comment:
I would prefer moving the array item type conversion down to the `transform` step - supplyDefaults (even the transform version) should try to avoid diving into arrays.